iree/compiler/Dialect/Flow/Conversion/TensorToFlow/ConvertTensorToFlow.cpp

```diff
@@ -319,20 +321,70 @@ struct ConvertTensorFromElementsPattern
   }
 };
 
+/// Converts linalg.tensor_reshape operations into flow.tensor.reshape
+/// operations.
+template <typename TensorReshapeOp>
+struct ConvertTensorReshapePattern : public OpRewritePattern<TensorReshapeOp> {
+  using OpRewritePattern<TensorReshapeOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(TensorReshapeOp reshapeOp,
+                                PatternRewriter &rewriter) const override {
+    if (reshapeOp->template getParentOfType<Flow::DispatchWorkgroupsOp>()) {
+      return failure();
+    }
+    SmallVector<SmallVector<Value>> outputShape;
+    ReifyRankedShapedTypeOpInterface reifyShapedTypeInterface =
+        cast<ReifyRankedShapedTypeOpInterface>(reshapeOp.getOperation());
+    if (failed(reifyShapedTypeInterface.reifyResultShapes(rewriter,
+                                                          outputShape))) {
+      return failure();
+    }
+    SmallVector<Value> outputDynamicShapes;
```

Comment:
Can we use `getDynamicDimValues()'?

Reply:
Actually this is resolving the shape of the output in terms of shape of its input. THis is done through the `ReifyRankedShapedTypeInterface` just above this code.